diagnostics/physics/ssh_eval.py

```diff
@@ -189,29 +108,43 @@ def plot_ssh_eval(pp_root, label):
         glorys_rg.where(mod_mask),
         grid[1],
         weights=model_grid.areacello.where(mod_mask).fillna(0),
-        x0=-89,
-        y0=48,
-        yint=2,
-        fontsize=8
+        fontsize=8,
+        x0=config['text_x'],
+        y0=config['text_y'],
+        xint=config['text_xint'],
+        plot_lat=config['plot_lat']
     )
+    logger.info("Successfully plotted glorys")
 
     for ax in grid:
-        add_ticks(ax, xlabelinterval=2, ylabelinterval=2, xticks=np.arange(-100, -31, 5), yticks=np.arange(0, 61, 5))
-        ax.set_xlim(xl, xr)
-        ax.set_ylim(yb, yt)
-        ax.set_xlabel('')
-        ax.set_ylabel('')
-        ax.set_facecolor('#bbbbbb')
-        for s in ax.spines.values():
-            s.set_visible(False)
+            # The set_xticks function called by add_ticks does not currently support non-rectangular projections.
+            # Until support is added, simply skip function if non-rectangular projection is used. Assume
+            # NorthPolarStereo will be the only non-rectangular projection inputted
+            if config['projection_grid'] != 'NorthPolarStereo':
+                add_ticks(ax, xlabelinterval=2, ylabelinterval=2,
+                          xticks=np.arange( config['lon']['west'], config['lon']['east'], 5 ),
+                          yticks=np.arange( config['lat']['south'], config['lat']['north'], 5 ),
+                          projection = proj
+                         )
+            else:
+                logger.warning("WARNING: Cannot set tick marks in non-rectangular projection. Will not call add_ticks")
```

Comment:
I’m okay with this approach. If we do want to add ticks to the `NorthPolarStereo` projection plot, we can try using `Gridliner`:

```
        # Add Gridliner for NorthPolarStereo projection
        gridliner = ax.gridlines(draw_labels=True, color='gray', linestyle='--')
        gridliner.xlabels_top = False  # Disable top labels
        gridliner.ylabels_right = False  # Disable right labels
        gridliner.xformatter = ccrs.cartopy.mpl.ticker.LongitudeFormatter()
        gridliner.yformatter = ccrs.cartopy.mpl.ticker.LatitudeFormatter()
        gridliner.xlabel_style = {'size': 10, 'color': 'gray'}
        gridliner.ylabel_style = {'size': 10, 'color': 'gray'}
        gridliner.xlocator = plt.FixedLocator([-180, -90, 0, 90, 180])  # Set longitudes
        gridliner.ylocator = plt.FixedLocator([60, 70, 80, 90])  # Set latitudes
```